validator/src/main.rs

```diff
@@ -2566,7 +2578,31 @@ pub fn main() {
         validator_config.max_ledger_shreds = Some(limit_ledger_size);
     }
 
+    // The size ratios between data shreds and coding shreds.
+    // The values will be used to determine the storage allocation
+    // for FIFO compaction.
+    let mut data_shred_size_ratio = 50;
+    let code_shred_size_ratio = 50;
+
     validator_config.blockstore_advanced_options = BlockstoreAdvancedOptions {
+        compression_type: match matches.value_of("ledger_compression") {
+            None => BlockstoreCompressionType::default(),
+            Some(ledger_compression_string) => {
+                // Adjust the data shred size percentage by assuming a
+                // 30% compression rate.
+                data_shred_size_ratio = 35;
```

Comment:
This PR also takes care of the scenario where both `ledger_compression` and `rocksdb_shred_compaction=fifo` are used and properly adjust the size allocation between data shreds and coding shreds.